fault/tester.py

```diff
@@ -536,6 +536,28 @@ def internal(self, *args):
         # return a SelectPath containing the desired path
         return SelectPath([self.circuit] + list(args))
 
+    def __call__(self, *args, **kwargs):
+        """
+        Poke the inputs of the circuit using *args (ordered, anonymous input
+        reference, excluding clocks) and **kwargs.  **kwargs will overwrite any
+        inputs written by *args.
+
+        Evaluate the circuit
+
+        Return the "peeked" output(s) of the circuit (tuple for multiple
+        outputs)
+        """
+        for arg, port in zip(args, self._circuit.interface.outputs()):
```

Comment:
Also this logic seems a bit weird...it expects `args` to be of length of # of inputs, then allows overriding using `kwargs`? Seems the expectation should be that `len(args) + len(kwargs) == num_inputs`...or just have a check that `len(args) <= num_inputs` and wire up only the first `m` inputs. E.g. see https://github.com/phanrahan/magma/blob/5bdb58bda377b363824629b719ddb0a4fb433fa8/magma/circuit.py#L391; having parity with that func. seems reasonable.

Reply:
zip will truncate by default, so you can use less inputs and it will skip the extra off the end (e.g. if they are wired up by kwargs

Reply:
I can add check that the total sums up to the expected amount, otherwise issue a warning

Reply:
The weird thing is, you can skip setting the clock, since it might be combinational evaluation, so actually that check might be overly conservative again

Reply:
So really it should be len(args) + len(kwargs that are not clock args)

Reply:
seems reasonable, the only thing is to avoid user confusion if either (a) a kwarg overrides an arg, or (b) a wrong number of total args are passed in (too few or too many)